scripts/weekly-summary

```diff
@@ -75,20 +76,49 @@ def get_artifacts(token, artifact_names, start, end):
       "expires_at": "2022-03-25T03:29:06Z"
     },
     """
-    req = Request(URL)
-    req.add_header("Accept", "application/vnd.github.v3+json")
-    req.add_header("Authorization", f"token {token}")
-    with urlopen(req) as r:
-        data = json.load(r)
 
-    # Filter out the artifacts within the date range and names
     artifacts = []
-    for a in data["artifacts"]:
-        if a["name"] not in artifact_names:
-            continue
-        updated_at = datetime.fromisoformat(a["updated_at"][:-1])
-        if start <= updated_at <= end:
-            artifacts.append(a)
+    page = 1
+    retry_limit = 3
+    while True:
+        req = Request(URL + f"&page={page}")
+        req.add_header("Accept", "application/vnd.github.v3+json")
+        req.add_header("Authorization", f"token {token}")
+        with urlopen(req) as r:
+            # Handle hitting the GitHub rate limit
+            # If the reset time is < 90s in the future, wait for it (trying 3 times)
+            # Otherwise raise an error
+            if r.status == 403:
+                try:
+                    reset = int(r.headers.get("X-RateLimit-Reset"))
+                except:
+                    raise RuntimeError("Hit GitHub rate limit. Reset header missing.")
+                if retry_limit == 0 or time.time() > reset or reset - time.time() > 90:
+                    raise RuntimeError("Hit GitHub rate limit. Reset is at %s" % time.ctime(reset))
+
+                # Try waiting until the reset time
+                time.sleep(reset - time.time())
```

Comment:
Maybe wait until a few moments later? Or is that practically guaranteed by the connections taking some time as well?

Reply:
You mean a few moments after the rest time? Yeah, I meant to do that...